src/linear_algebra/dual.lean

```diff
@@ -752,3 +753,90 @@ end
 end field
 
 end linear_map
+
+section tensor_product
+
+variables {M : Type*} {N : Type*}
+variables [add_comm_group M] [add_comm_group N]
+variables [module R M] [module R N]
+
+variables {ι κ : Type*}
+variables [decidable_eq ι] [decidable_eq κ]
+variables [fintype ι] [fintype κ]
+
+open_locale big_operators
+open_locale tensor_product
+
+local attribute [ext] tensor_product.ext
+
+open tensor_product
+open linear_map
+open module (dual)
+
+/--
+The canonical linear map from `dual M ⊗ dual N` to `dual (M ⊗ N)`,
+sending `f ⊗ g` to the composition of `tensor_product.map f g` with
+the natural isomorphism `R ⊗ R ≃ R`.
+-/
+def dual_tensor_dual_map : (dual R M) ⊗[R] (dual R N) →ₗ[R] dual R (M ⊗[R] N) :=
+(comp_right ↑(tensor_product.lid R R)) ∘ₗ hom_tensor_hom_map
+
+@[simp]
+lemma dual_tensor_dual_map_apply (f : dual R M) (g : dual R N) (m : M) (n : N) :
+  dual_tensor_dual_map (f ⊗ₜ g) (m ⊗ₜ n) = f m * g n :=
+by simp only [dual_tensor_dual_map, coe_comp, function.comp_app, hom_tensor_hom_map_apply,
+  comp_right_apply, linear_equiv.coe_coe, map_tmul, lid_tmul, algebra.id.smul_eq_mul]
+
+/--
+An inverse to `dual_tensor_dual_map` given bases.
+-/
+noncomputable
+def dual_tensor_dual_inv_of_basis (b : basis ι R M) (c : basis κ R N) :
+  dual R (M ⊗[R] N) →ₗ[R] (dual R M) ⊗[R] (dual R N) :=
+∑ i j, (ring_lmap_equiv_self R ℕ _).symm (b.dual_basis i ⊗ₜ c.dual_basis j)
+    ∘ₗ applyₗ (c j) ∘ₗ applyₗ (b i) ∘ₗ (lcurry R M N R)
```

Comment:
Operators (like ∘ₗ) usually goes to the end of the previous line instead of the head of the next line.